pyanaconda/core/configuration/anaconda.py

```diff
@@ -403,6 +403,11 @@ def set_from_opts(self, opts):
         if opts.noverifyssl:
             self.payload._set_option("verify_ssl", not opts.noverifyssl)
 
+        # Set geolocation provider
+        # FIXME: This will be removed once the boot option becomes a boolean
+        if "geoloc" in opts and opts.geoloc:
+            self.ui._set_option("geolocation_provider", opts.geoloc)
```

Comment:
Could you set here `opts.geoloc.removeprefix("provider_")`? If you do that and update the `GEOLOC_PROVIDER_*` constants, it might be enough to get rid of that useless prefix in the configuration files.

Reply:
I'd like to leave that after the general structure is worked out, but yes, let's have it.

Reply:
If I make the conf values use only URLs, then removing the prefix becomes meaningless, it will go away eventually anyway.